core/src/main/java/org/apache/brooklyn/core/mgmt/internal/BrooklynShutdownHooks.java

```diff
@@ -100,6 +99,17 @@ public static void invokeStopOnShutdown(Entity entity) {
         }
     }
 
+    public static void resetShutdownFlag() {
```

Comment:
I guess you'd need both boolean parameters for the `compareAndSet`.

It's just a suggestion, matter of taste really, certainly not needed as such.